### PR TITLE
add method for MOI.EqualTo{Float64} set

### PR DESCRIPTION
**Pull request by jd-lara**
The previous implementation called `Lib.XPRSchgbounds(), Ref(value))` twice once for L and once for B. This implied that when calling `JuMP.fix()` Xpress would throw warnings if a previous UB or LB was above or below the new value. In large model it would generate a lot of noise in the logging. 

This PR employs `Lib.XPRSchgbounds(model.inner, Cint(1), Ref(Cint(info.column-1)), Ref(UInt8('B')), Ref(value))` in a new method for attempts to fix the variable and remove the warning calls.

### CODE REVIEW

File: src/MOI/MOI_wrapper.jl [hunk 1]
```diff
@@ -1484,11 +1484,18 @@ function MOI.delete(
...
+    info.previous_upper_bound = _get_variable_upper_bound(model, info)
+    return
+end
+
 """
     _set_variable_lower_bound(model, info, value)
 
 This function is used to indirectly set the lower bound of a variable.
-
+    MathOptInterface.EqualTo{Float64}
```
Comment (odow) [inline src/MOI/MOI_wrapper.jl:1498]:
```suggestion
```

File: src/MOI/MOI_wrapper.jl [hunk 2]
```diff
@@ -1672,6 +1679,20 @@ function MOI.get(
...
+    model::Optimizer,
+    ::MOI.ConstraintSet,
+    c::MOI.ConstraintIndex{MOI.VariableIndex, MOI.EqualTo{Float64}},
+    s::MOI.EqualTo{Float64}
+)
+    MOI.throw_if_not_valid(model, c)
+    lower, upper = _bounds(s)
+    @assert lower == upper
+    info = _info(model, c)
+    _fix_variable_value(model, info, lower)
```
Comment (odow) [inline src/MOI/MOI_wrapper.jl:1692]: Instead of introducing this new function, just put the `XPRSchgbounds` call here directly.

Also work a comment that `XPRSchgbounds` expects vectors, so we pass singular `Ref`s, and that `B` stands for "both" bounds.

File: src/MOI/MOI_wrapper.jl [hunk 3]
```diff
@@ -1484,11 +1484,18 @@ function MOI.delete(
     return
 end
 
+function _fix_variable_value(model, info, value)
+    Lib.XPRSchgbounds(model.inner, Cint(1), Ref(Cint(info.column-1)), Ref(UInt8('B')), Ref(value))
```
Comment (odow) [inline src/MOI/MOI_wrapper.jl:1488]:
```suggestion
    Lib.XPRSchgbounds(model.inner, Cint(1), Ref(Cint(info.column-1)), Ref(Cchar('B')), Ref(value))
```

Comment (jd-lara) [reply src/MOI/MOI_wrapper.jl:1488]: This change causes an error: `unsafe_convert(::Type{Ptr{UInt8}}, ::Base.RefValue{Int8})`

File: src/MOI/MOI_wrapper.jl (hunk 2 above)
Comment (jd-lara) [reply src/MOI/MOI_wrapper.jl]: Fixed via [24c98d4](https://github.com/jump-dev/Xpress.jl/pull/185/commits/24c98d4f7cd0b4af16917273092d7896060e7379)

**Comment by joaquimg** [comment]
I have been talking about similar things with @guilhermebodin , lets us take a look.

File: src/MOI/MOI_wrapper.jl (hunk 3 above)
Comment (odow) [reply src/MOI/MOI_wrapper.jl]: Urgh. Okay. The C documentation says it should be a `Cchar`: https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/XPRSchgbounds.html. I don't know how the `lib.jl` was generated. Perhaps the header file is actually a UInt8 instead of Int8.

**Comment by jd-lara** [comment]
FYI the tests are failing due to: `ERROR: LoadError: InitError: Xpress licensing error 10: Your license expired on 31-oct-2022. Please contact support@fico.com to renew it.`

**Comment by joaquimg** [comment]
The main issue is that `_set_variable_lower_bound` doe more than just calling `XPRSchgbounds`.
So we need to protect against those cases because the equality change end up changing the LB.
Needs something more carefully written.
Also if tests passed locally, the are not capturing this corner case that might require a model with a SOC.

**Comment by joaquimg** [comment]
> FYI the tests are failing due to: ERROR: LoadError: InitError: Xpress licensing error 10: Your license expired on 31-oct-2022. Please contact support@fico.com to renew it.

Thank we will update.

**Comment by jd-lara** [comment]
> The main issue is that `_set_variable_lower_bound` doe more than just calling `XPRSchgbounds`.
> So we need to protect against those cases because the equality change end up changing the LB.
> Needs something more carefully written.
> Also if tests passed locally, the are not capturing this corner case that might require a model with a SOC.

We can add the check let me try something.

**Comment by jd-lara** [comment]
@joaquimg check this commit [7ca6e4b](https://github.com/jump-dev/Xpress.jl/pull/185/commits/7ca6e4b9d4be21fe7a096630a3c49b1fbd9fbe52)

**Comment by jd-lara** [comment]
local tests outcomes 

image

**Comment by jd-lara** [comment]
bump to @odow and @joaquimg.

**Review by odow — APPROVED** [review]
Seem okay. @joaquimg is the expert on Xpress though.

**Comment by joaquimg** [comment]
on it!

**Comment by joaquimg** [comment]
replaced by #187
